packages/components/src/components/pin-code/pin-code.tsx

```diff
@@ -0,0 +1,267 @@
+import {
+  initHiddenInput,
+  inputIsSingleDigit,
+  joinInputValues,
+  PIN_CODE_LENGTHS,
+  PIN_CODE_TYPES,
+  syncHiddenInput,
+} from './pin-code-utils';
+import { StateMessage } from '../common/state-message/state-message';
+import { Required } from '../common/required/required';
+
+const propTypes: PropTypes<typeof PinCode> = {
+  label: AllowedTypes.string,
+  description: AllowedTypes.string,
+  name: AllowedTypes.string,
+  length: AllowedTypes.oneOf<PinCodeLength>(PIN_CODE_LENGTHS),
+  hideLabel: AllowedTypes.breakpoint('boolean'),
+  state: AllowedTypes.oneOf<PinCodeState>(FORM_STATES),
+  disabled: AllowedTypes.boolean,
+  loading: AllowedTypes.boolean,
+  required: AllowedTypes.boolean,
+  message: AllowedTypes.string,
+  type: AllowedTypes.oneOf<PinCodeType>(PIN_CODE_TYPES),
+  value: AllowedTypes.oneOf<ValidatorFunction>([AllowedTypes.string, AllowedTypes.number]),
+  theme: AllowedTypes.oneOf<Theme>(THEMES),
+};
+
+@Component({
+  tag: 'p-pin-code',
+  shadow: true,
+})
+export class PinCode {
+  @Element() public host!: HTMLElement;
+
+  /** The label text. */
+  @Prop() public label?: string = '';
+
+  /** The description text. */
+  @Prop() public description?: string = '';
+
+  /** Name of the control. */
+  @Prop() public name?: string;
+
+  /** Number of characters of the Pin Code. */
+  @Prop() public length?: PinCodeLength = 4;
+
+  /** Show or hide label and description text. For better accessibility it is recommended to show the label. */
+  @Prop() public hideLabel?: BreakpointCustomizable<boolean> = false;
+
+  /** The validation state. */
+  @Prop() public state?: PinCodeState = 'none';
+
+  /** Disables the Pin Code. No events will be triggered while disabled state is active. */
+  @Prop() public disabled?: boolean = false;
+
+  /** Disables the Pin Code and shows a loading indicator. No events will be triggered while loading state is active. */
+  @Prop() public loading?: boolean = false;
+
+  /** Marks the Pin Code as required. */
+  @Prop() public required?: boolean = false;
+
+  /** The message styled depending on validation state. */
+  @Prop() public message?: string = '';
+
+  /** Pin Code type. */
+  @Prop() public type?: PinCodeType = 'number';
+
+  /** Sets the initial value of the Pin Code. */
+  @Prop({ mutable: true }) public value?: string = '';
+
+  /** Adapts the color depending on the theme. */
+  @Prop() public theme?: Theme = 'light';
+
+  /** Emitted when selected element changes. */
+  @Event({ bubbles: false }) public update: EventEmitter<PinCodeUpdateEvent>;
+
+  private isWithinForm: boolean;
+  private hiddenInput: HTMLInputElement;
+  private pinCodeElements: HTMLInputElement[] = [];
+
+  public connectedCallback(): void {
+    this.isWithinForm = isWithinForm(this.host);
+  }
+
+  public componentWillLoad(): void {
+    // make sure initial value is not longer than pin code length
+    if (this.value) {
+      this.value = this.value.slice(0, this.length);
```

Comment:
Inform the user that the value is longer than the length with consoleWarn or maybe even throw an error instead of slicing the value